csrc/kernel_db/kernel_db.cpp

```diff
@@ -223,46 +223,50 @@ bool KernelDb::write(
     const std::vector<char>& cubin) {
   FUSER_PERF_SCOPE("KernelDb::write");
   std::lock_guard<std::mutex> guard(kernel_db_lock);
-  bool status = false;
+
+  // Short-circuit path if kernel already exist in database.
+  // Only return false if it does not already exist in the database and we fail
+  // to add kernel to database.
+  if (kernel_map_.count(kernel_code) > 0) {
```

Comment:
This is the only real change that the status boolean and outer conditional was removed in favor of short circuit.